docs/_posts/2021-02-13-hudi-key-generators.md

```diff
@@ -0,0 +1,139 @@
+
+### [TimestampBasedKeyGenerator](https://github.com/apache/hudi/blob/master/hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/TimestampBasedKeyGenerator.java)
+This key generator relies on timestamps for the partition field. The field values are interpreted as timestamps 
+and not just converted to string while generating partition path value for records.  Record key is same as before where it is chosen by 
+field name.  Users are expected to set few more configs to use this KeyGenerator.
+
+Configs to be set:
+
+| Config        | Meaning/purpose |       
+| ------------- | -------------|
+| ```hoodie.deltastreamer.keygen.timebased.timestamp.type```    | One of the timestamp types supported(UNIX_TIMESTAMP, DATE_STRING, MIXED, EPOCHMILLISECONDS, SCALAR) |
+| ```hoodie.deltastreamer.keygen.timebased.output.dateformat```| Output date format | 
+| ```hoodie.deltastreamer.keygen.timebased.timezone```| Timezone of the data format| 
+| ```oodie.deltastreamer.keygen.timebased.input.dateformat```| Input date format |
+
+Lets go over some example values for TimestampBasedKeyGenerator.
+
+<br/>
+// Timestamp is GMT
+
+| Config field | Value |
+| ------------- | -------------|
+|```hoodie.deltastreamer.keygen.timebased.timestamp.type```| "EPOCHMILLISECONDS"|
+|```hoodie.deltastreamer.keygen.timebased.output.dateformat``` | "yyyy-MM-dd hh" |
+|```hoodie.deltastreamer.keygen.timebased.timezone```| "GMT+8:00" |
+
+Input Field value: “1578283932000L”
+Partition path generated from key generator: “2020-01-06 12”
+
+If input field value is null for some rows.
+Partition path generated from key generator: “1970-01-01 08”
+
+<br/>
+// Timestamp is DATE_STRING
+
+| Config field | Value |
+| ------------- | -------------|
+|```hoodie.deltastreamer.keygen.timebased.timestamp.type```|  "DATE_STRING"  |
+|```hoodie.deltastreamer.keygen.timebased.output.dateformat```|  "yyyy-MM-dd hh" | 
+|```hoodie.deltastreamer.keygen.timebased.timezone```|  "GMT+8:00" |
+|```hoodie.deltastreamer.keygen.timebased.input.dateformat```|  "yyyy-MM-dd hh:mm:ss" |
+
+Input field value: “2020-01-06 12:12:12”
+Partition path generated from key generator: “2020-01-06 12”
+
+If input field value is null for some rows.
+Partition path generated from key generator: “1970-01-01 12:00:00”
+<br/>
+<br/>
+
+// Scalar examples
+
+| Config field | Value |
+| ------------- | -------------|
+|```hoodie.deltastreamer.keygen.timebased.timestamp.type```| "SCALAR"|
+|```hoodie.deltastreamer.keygen.timebased.output.dateformat```| "yyyy-MM-dd hh" |
+|```hoodie.deltastreamer.keygen.timebased.timezone```| "GMT" |
+|```hoodie.deltastreamer.keygen.timebased.timestamp.scalar.time.unit```| "days" |
+
+Input field value: “20000L”
+Partition path generated from key generator: “2024-10-04 12”
+
+If input field value is null.
+Partition path generated from key generator: “1970-01-02 12”
+
+// More to be filled in.
```

Comment:
@pratyakshsharma : if you can fill in these details, we can open it up for review.